src/temp_sensor.rs

```diff
@@ -0,0 +1,165 @@
+#[cfg(any(esp32s2, esp32s3, esp32c2, esp32c3, esp32c6, esp32h2))]
+use esp_idf_sys::{
+    soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_DEFAULT,
+    soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST,
+    soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL,
+    temperature_sensor_clk_src_t, temperature_sensor_config_t, temperature_sensor_disable,
+    temperature_sensor_enable, temperature_sensor_get_celsius, temperature_sensor_handle_t,
+    temperature_sensor_install, temperature_sensor_uninstall,
+};
+
+// -- TemperatureSensorClockSource --
+
+#[derive(Copy, Clone)]
+#[cfg(any(esp32s2, esp32s3, esp32c2, esp32c3, esp32c6, esp32h2))]
```

Comment:
to make the #[cfg(any(esp32s2, esp32s3, esp32c2, esp32c3, esp32c6, esp32h2))] not so verbose i think its currently best to list the "not any" - so the original esp32 and so forth, because this list will potentially only grow and the chips that not support it will be more likely the same

Reply:
Done in bb519ff751c16191eebb1f54592bd8ed43b72d68